core/templates/dev/head/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -631,30 +631,33 @@ angular.module('oppia').directive('questionsList', [
                       return;
                     }
 
-                    var modalInstance = $uibModal.open({
-                      templateUrl:
-                             UrlInterpolationService.getDirectiveTemplateUrl(
-                               '/components/question-directives' +
-                               '/modal-templates/' +
-                               'question-editor-save-modal.template.html'),
-                      backdrop: true,
-                      controller: [
-                        '$scope', '$uibModalInstance',
-                        function($scope, $uibModalInstance) {
-                          $scope.save = function(commitMessage) {
-                            $uibModalInstance.close(commitMessage);
-                          };
-                          $scope.cancel = function() {
-                            $uibModalInstance.dismiss('cancel');
-                          };
-                        }
-                      ]
-                    });
-
-                    modalInstance.result.then(function(commitMessage) {
-                      returnModalObject.commitMessage = commitMessage;
+                    if (QuestionUndoRedoService.hasChanges()) {
```

Comment:
The way this code is written, it's a bit confusing because it sounds like $scope.saveAndCommit would be called sometimes in cases where QuestionUndoRedoService.hasChanges() is false (which seems counterintuitive -- why would we allow saving/committing something when there are no changes)?

Could you please add a clear docstring to saveAndCommit() to explain when it should be called (and while you're at it, change the name to follow lowerCamelCase correctly -- "commit" should start with a "C")?

Reply:
done

Reply:
> it's a bit confusing because it sounds like $scope.saveAndCommit would be called sometimes in cases where QuestionUndoRedoService.hasChanges() is false (which seems counterintuitive -- why would we allow saving/committing something when there are no changes)?

 @seanlip just for clarification, `QuestionUndoRedoService.hasChanges()`  can find only the changes in the content of the question, but cannot detect the changes in skill linkages.